lib/pf/admin_roles.pm

```diff
@@ -18,113 +18,13 @@ use warnings;
 use base qw(Exporter);
 use pf::file_paths;
 use List::MoreUtils qw(any all uniq);
-use pf::config::cached;
+use pfconfig::cached_hash;
+use pf::constants::admin_roles;
 
-our @EXPORT = qw(admin_can admin_can_do_any admin_can_do_any_in_group @ADMIN_ACTIONS %ADMIN_ROLES $cached_adminroles_config admin_allowed_options admin_allowed_options_all);
+our @EXPORT = qw(admin_can admin_can_do_any admin_can_do_any_in_group @ADMIN_ACTIONS %ADMIN_ROLES admin_allowed_options admin_allowed_options_all);
 our %ADMIN_ROLES;
-our @ADMIN_ACTIONS = qw(
-    ADMIN_ROLES_CREATE
-    ADMIN_ROLES_DELETE
-    ADMIN_ROLES_READ
-    ADMIN_ROLES_UPDATE
-
-    CONFIGURATION_MAIN_READ
-    CONFIGURATION_MAIN_UPDATE
-
-    FINGERPRINTS_READ
-    FINGERPRINTS_UPDATE
-
-    FIREWALL_SSO_READ
-    FIREWALL_SSO_CREATE
-    FIREWALL_SSO_UPDATE
-    FIREWALL_SSO_DELETE
-
-    FLOATING_DEVICES_CREATE
-    FLOATING_DEVICES_DELETE
-    FLOATING_DEVICES_READ
-    FLOATING_DEVICES_UPDATE
-
-    INTERFACES_CREATE
-    INTERFACES_DELETE
-    INTERFACES_READ
-    INTERFACES_UPDATE
-
-    MAC_READ
-    MAC_UPDATE
-
-    NODES_CREATE
-    NODES_DELETE
-    NODES_READ
-    NODES_UPDATE
-
-    PORTAL_PROFILES_CREATE
-    PORTAL_PROFILES_DELETE
-    PORTAL_PROFILES_READ
-    PORTAL_PROFILES_UPDATE
-
-    PROVISIONING_CREATE
-    PROVISIONING_DELETE
-    PROVISIONING_READ
-    PROVISIONING_UPDATE
-
-    REPORTS
-    SERVICES
-
-    SOH_CREATE
-    SOH_DELETE
-    SOH_READ
-    SOH_UPDATE
-
-    SWITCHES_CREATE
-    SWITCHES_DELETE
-    SWITCHES_READ
-    SWITCHES_UPDATE
-
-    USERAGENTS_READ
-
-    USERS_READ
-    USERS_CREATE
-    USERS_UPDATE
-    USERS_DELETE
-    USERS_SET_ROLE
-    USERS_SET_ACCESS_DURATION
-    USERS_SET_UNREG_DATE
-    USERS_SET_ACCESS_LEVEL
-    USERS_MARK_AS_SPONSOR
-
-    USERS_ROLES_CREATE
-    USERS_ROLES_DELETE
-    USERS_ROLES_READ
-    USERS_ROLES_UPDATE
-
-    USERS_SOURCES_CREATE
-    USERS_SOURCES_DELETE
-    USERS_SOURCES_READ
-    USERS_SOURCES_UPDATE
-
-    VIOLATIONS_CREATE
-    VIOLATIONS_DELETE
-    VIOLATIONS_READ
-    VIOLATIONS_UPDATE
-    SOH_READ
-    SOH_CREATE
-    SOH_UPDATE
-    SOH_DELETE
-    FINGERPRINTS_READ
-    FINGERPRINTS_UPDATE
-    USERAGENTS_READ
-    MAC_READ
-    MAC_UPDATE
-    FIREWALL_SSO_READ
-    FIREWALL_SSO_CREATE
-    FIREWALL_SSO_UPDATE
-    FIREWALL_SSO_DELETE
-    REALM_READ
-    REALM_CREATE
-    REALM_UPDATE
-    REALM_DELETE
-);
-
+tie %ADMIN_ROLES, 'pfconfig::cached_hash', 'config::AdminRoles';
+our @ADMIN_ACTIONS = @pf::constants::admin_roles::ADMIN_ACTIONS;
```

Comment:
Not sure what you mean. I personally find it ok this way but I'm open to making it different